src/ct_eval.cpp

```diff
@@ -94,15 +96,15 @@ TypedAST::TernaryExpression* ct_eval(
 	return ast;
 }
 
-TypedAST::RecordAccessExpression* ct_eval(
-    TypedAST::RecordAccessExpression* ast, TypeChecker& tc, TypedAST::Allocator& alloc) {
-	ast->m_record = ct_eval(ast->m_record, tc, alloc);
+TypedAST::AccessExpression* ct_eval(
+    TypedAST::AccessExpression* ast, TypeChecker& tc, TypedAST::Allocator& alloc) {
+	ast->m_object = ct_eval(ast->m_object, tc, alloc);
 	return ast;
 }
```

Comment:
It would be cool if this returned a `TypedAST::Constructor*` when the metatype is `constructor`

Reply:
I feel like this is more consistent with the idea that `ct_eval` returns the compile time value of an expression

Reply:
I did not think about this (I always forget about that). I will add a test and try to make it work then, so don't worry about merging for now

Reply:
I will look into it.